src/implementations/process/spin_pols.jl

```diff
@@ -0,0 +1,144 @@
+
+@inline _multiplicity(::AbstractDefinitePolarization) = 1
+@inline _multiplicity(::AbstractDefiniteSpin) = 1
+@inline _multiplicity(::AllSpin) = 2
+@inline _multiplicity(::AllPol) = 2
+
+@inline _multiplicity(::SyncedPolarization{N}, seen_spin_pols::Tuple{}) where {N} = 2
+@inline _multiplicity(::SyncedSpin{N}, seen_spin_pols::Tuple{}) where {N} = 2
+
+@inline function _multiplicity(
+    sp::SyncedPolarization{N}, seen_pols::Tuple{Int,Vararg{Int}}
+) where {N}
+    if seen_pols[1] == N
+        return 1
+    else
+        return _multiplicity(sp, seen_pols[2:end])
+    end
+end
+
+@inline function _multiplicity(
+    ss::SyncedSpin{N}, seen_spins::Tuple{Int,Vararg{Int}}
+) where {N}
+    if seen_spins[1] == N
+        return 1
+    else
+        return _multiplicity(ss, seen_spins[2:end])
+    end
+end
+
+# multiplicity of the empty spin_pol set is 1
+@inline _multiplicity(::Tuple{}, _, _) = 1
+
+# recursion for abstract spins or pols that are not synced
+@inline function _multiplicity(
+    spin_pols::Tuple{AbstractSpin,Vararg{AbstractSpinOrPolarization}},
+    seen_spins::NTuple{S,Int},
+    seen_pols::NTuple{P,Int},
+) where {S,P}
+    return _multiplicity(spin_pols[1]) *
+           _multiplicity(spin_pols[2:end], seen_spins, seen_pols)
+end
+@inline function _multiplicity(
+    spin_pols::Tuple{AbstractPolarization,Vararg{AbstractSpinOrPolarization}},
+    seen_spins::NTuple{S,Int},
+    seen_pols::NTuple{P,Int},
+) where {S,P}
+    return _multiplicity(spin_pols[1]) *
+           _multiplicity(spin_pols[2:end], seen_spins, seen_pols)
+end
+
+# recursion for synced spins or pols
+@inline function _multiplicity(
+    spin_pols::Tuple{SyncedPolarization{N},Vararg{AbstractSpinOrPolarization}},
+    seen_spins::NTuple{S,Int},
+    seen_pols::NTuple{P,Int},
+) where {N,S,P}
+    return _multiplicity(spin_pols[1], seen_pols) *
+           _multiplicity(spin_pols[2:end], seen_spins, (seen_pols..., N))
+end
+@inline function _multiplicity(
+    spin_pols::Tuple{SyncedSpin{N},Vararg{AbstractSpinOrPolarization}},
+    seen_spins::NTuple{S,Int},
+    seen_pols::NTuple{P,Int},
+) where {N,S,P}
+    return _multiplicity(spin_pols[1], seen_spins) *
+           _multiplicity(spin_pols[2:end], (seen_spins..., N), seen_pols)
+end
+
+"""
+    multiplicity(proc::AbstractProcessDefinition)
+
+Return the number of spin and polarization combinations represented by `proc` total. This depends on the specific [`AbstractSpinOrPolarization`](@ref)s returned by [`spin_pols`](@ref) for `proc`.
+For example, a default Compton process with four indefinite spins/polarizations has a multiplicity of 2^4 = 16. A Compton process with many incoming photons that have synced polarizations
+will still have a multiplicity of 16.
+
+!!! note Performance
+    As long as [`incoming_spin_pols`](@ref) and [`outgoing_spin_pols`](@ref) can be evaluated at compile time, this function is completely compiled away.
```

Comment:
At some point, we should add such statements to the docs of the respective function as well.

Reply:
I added small notes to the `AbstractProcessDefinition` interface